acorn/src/expression.js

```diff
@@ -237,6 +237,9 @@ pp.parseMaybeUnary = function(refDestructuringErrors, sawUnary) {
     else if (this.strict && node.operator === "delete" &&
              node.argument.type === "Identifier")
       this.raiseRecoverable(node.start, "Deleting local variable in strict mode")
+    else if (node.operator === "delete" && node.argument.type === "MemberExpression" &&
+             node.argument.property.type === "PrivateIdentifier")
+      this.raiseRecoverable(node.start, "Private fields can not be deleted")
```

Comment:
Private optional property can not be deleted.

```js
testFail(`class Foo {
  #x;
  constructor() {
    delete this?.#x;
  }
}
`, "", {ecmaVersion: 13})
```

Reply:
Fixed in https://github.com/acornjs/acorn/pull/1026/commits/464858c92dad71887aed46352cdc2381680bf324.

Also fixed in https://github.com/acornjs/acorn/pull/1026/commits/24aee3780ec119417d4251b69da559de8a6e1aef about `new.target`.